solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,679 @@
+  type        = string
+  default     = null
+  description = "The CRN of the existing key management service (KMS) that is used to create keys for encrypting the flow logs Cloud Object Storage bucket."
+}
+
+variable "kms_endpoint_type" {
+  type        = string
+  description = "The type of endpoint to use for communicating with the KMS. Possible values: `public`, `private`. Applies only if `existing_flow_logs_bucket_kms_key_crn` is not specified."
+  default     = "private"
+  validation {
+    condition     = can(regex("public|private", var.kms_endpoint_type))
+    error_message = "Valid values for the `kms_endpoint_type_value` are `public` or `private`."
+  }
+}
+
+variable "kms_key_ring_name" {
+  type        = string
+  default     = "flow-logs-cos-key-ring"
+  description = "The name of the key ring to create for the Cloud Object Storage bucket key. If an existing key is used, this variable is not required. If the prefix input variable is passed, the name of the key ring is prefixed to the value in the `<prefix>-value` format."
+}
+
+variable "kms_key_name" {
+  type        = string
+  default     = "flow-logs-cos-key"
+  description = "The name of the key to encrypt the flow logs Cloud Object Storage bucket. If an existing key is used, this variable is not required. If the prefix input variable is passed, the name of the key is prefixed to the value in the `<prefix>-value` format."
+}
+
+##############################################################################
+# Optional VPC Variables
+##############################################################################
+
+variable "default_network_acl_name" {
+  description = "Name of the Default ACL. If null, a name will be automatically generated."
+  type        = string
+  default     = null
+}
+
+variable "default_security_group_name" {
+  description = "Name of the Default Security Group. If null, a name will be automatically generated."
+  type        = string
+  default     = null
+}
+
+variable "default_routing_table_name" {
+  description = "Name of the Default Routing Table. If null, a name will be automatically generated."
+  type        = string
+  default     = null
+}
+
+##############################################################################
+# VPN Gateways
+##############################################################################
+
+variable "vpn_gateways" {
+  description = "List of VPN Gateways to create."
+  type = list(
+    object({
+      name           = string
+      vpc_name       = string
+      subnet_name    = string # Do not include prefix, use same name as in `var.subnets`
+      mode           = optional(string)
+      resource_group = optional(string)
+      access_tags    = optional(list(string), [])
+    })
+  )
+
+  default = []
+}
+
+##############################################################################
+# VPE Gateways
+##############################################################################
+
+variable "cloud_services" {
```

Comment:
suggest to prefix this with `vpe_gateway_`

Reply:
updated